symphony/bdk/core/service/connection/connection_service.py

```diff
@@ -0,0 +1,133 @@
+from symphony.bdk.core.auth.auth_session import AuthSession
+from symphony.bdk.core.service.connection.model.connection_status import ConnectionStatus
+from symphony.bdk.gen.pod_api.connection_api import ConnectionApi
+from symphony.bdk.gen.pod_model.user_connection import UserConnection
+from symphony.bdk.gen.pod_model.user_connection_request import UserConnectionRequest
+
+
+class ConnectionService:
+    """Service class for managing the connections between users
+
+    This service is used for retrieving the connection status if the calling user with a specified user or with many
+    other internal or external users in the pod, and perform some actions related to the connection status like:
+
+    * Send a connection request to an user
+    * Accept a connection request from an user
+    * Reject a connection request from an user
+    * Remove a connection with an user
+    """
+
+    def __init__(self, connection_api: ConnectionApi, auth_session: AuthSession):
+        self._connection_api = connection_api
+        self._auth_session = auth_session
+
+    async def get_connection(self, user_id: int) -> UserConnection:
+        """
+        Get connection status, i.e. check if the calling user is connected to the specified user.
+        See: `Get Connection <https://developers.symphony.com/restapi/reference#get-connection>`_
+
+        :param user_id: The id of the user with whom the caller want to check.
+
+        :return: Connection status with the specified user.
+
+        """
+        params = {
+            'user_id': str(user_id),
+            'session_token': await self._auth_session.session_token
+        }
+        return await self._connection_api.v1_connection_user_user_id_info_get(**params)
+
+    async def list_connections(
+            self,
+            status: ConnectionStatus = ConnectionStatus.ALL,
+            user_ids: [int] = None
+    ) -> [UserConnection]:
+        """
+        List all current connection statuses with external or specified users.
```

Comment:
I would perhaps phrase it as List connections of the requesting users with...

To me it was not obvious if we listing the connections of the user making the call or of the users passed as parameters to the call